cloud/services/computes/tags.go

```diff
@@ -0,0 +1,76 @@
+package computes
+
+import (
+	"github.com/digitalocean/godo"
+	"k8s.io/apimachinery/pkg/api/errors"
+	infrav1 "sigs.k8s.io/cluster-api-provider-digitalocean/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-digitalocean/cloud/scope"
+	"strconv"
+)
+
+func (s *Service) Retag(droplet *godo.Droplet, scope *scope.MachineScope) error {
+	tags := infrav1.BuildTags(infrav1.BuildTagParams{
+		ClusterName: infrav1.DOSafeName(s.scope.Name()),
+		ClusterUID:  s.scope.UID(),
+		Name:        infrav1.DOSafeName(scope.Name()),
+		Role:        scope.Role(),
+		Additional:  scope.AdditionalTags(),
+	})
+
+	for _, tag := range tags {
+		if !contains(droplet.Tags, tag) {
+			if err := s.createTag(tag); err != nil {
+				return err
+			}
+			_, err := s.scope.Tags.TagResources(s.ctx, tag, &godo.TagResourcesRequest{
+				Resources: []godo.Resource{
+					{
+						ID:   strconv.Itoa(droplet.ID),
+						Type: godo.DropletResourceType,
+					},
+				},
+			})
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// do we ever add tags to CPC droplets in another way or manually that might be deleted by the below logic?
```

Comment:
@timoreimann might know this answer best ^

Reply:
I think we can safely assume full management of droplets that capdo creates. In general they shouldn't be touched manually.
But what are those "CPC droplets" you speak of? 😜